service.go

```diff
@@ -1595,94 +1595,83 @@ func (s *Service) GetPayloadTrusted(ctx context.Context, log *zerolog.Logger, in
-
+	_, prefetchPayloadToSignedBlindedBeaconBlockSpan := s.tracer.Start(ctx, "getPayload-prefetchPayloadToSignedBlindedBeaconBlockSpan")
 	blindedBeaconBlock, errRes := s.prefetchPayloadToSignedBlindedBeaconBlock(ctx, in.Payload)
 	if errRes != nil {
 		log.Error().Err(errRes).Msg("prefetchPayloadToSignedBlindedBeaconBlock failed")
 		go s.sendPayloadStats(in.Payload, log, false, nil, in.ReceivedAt, startTime, time.Now(), 0, id, in.ClientIP, in.ValidatorID, in.AccountID, latency, in.Cluster, in.UserAgent, in.SlotUID)
 		return nil, errRes
 	}
+	prefetchPayloadToSignedBlindedBeaconBlockSpan.End()
 
-	slot, err := blindedBeaconBlock.Slot()
-	if err != nil {
-		return nil, toErrorResp(http.StatusBadRequest, "failed to get slot")
-	}
-	blockHash, err := blindedBeaconBlock.ExecutionBlockHash()
-	if err != nil {
-		return nil, toErrorResp(http.StatusBadRequest, "failed to get block hash")
-	}
-	parentHash, err := blindedBeaconBlock.ExecutionParentHash()
-	if err != nil {
-		return nil, toErrorResp(http.StatusBadRequest, "failed to get parent hash")
-	}
-	slotInt := int64(slot)
-	blockHashStr := blockHash.String()
-	parentHashStr := parentHash.String()
-	uKey := fmt.Sprintf("slot_%v_bHash_%v_pHash_%v", slotInt, blockHashStr, parentHashStr)
-
-	parentSpan.SetAttributes(
-		attribute.Int64("slot", slotInt),
-		attribute.String("blockHash", blockHashStr),
-		attribute.String("parentHash", parentHashStr),
-		attribute.String("uniqueKey", uKey),
-	)
-	*log = log.With().
-		Int64("slot", slotInt).
-		Str("blockHash", blockHashStr).
-		Str("parentHash", parentHashStr).
-		Str("uniqueKey", uKey).
-		Logger()
-
-	payloadInfo, errRes := s.validateAndFetchPayload(ctx, blindedBeaconBlock)
-	if errRes != nil {
-		// TODO: wait for relay response
-		go s.sendPayloadStats(in.Payload, log, false, payloadInfo, in.ReceivedAt, startTime, time.Now(), 0, id, in.ClientIP, in.ValidatorID, in.AccountID, latency, in.Cluster, in.UserAgent, in.SlotUID)
-		return nil, errRes
-	}
-
-	// return immediately
-	slotStartTime := GetSlotStartTime(s.beaconGenesisTime, int64(payloadInfo.Slot), s.secondsPerSlot)
-	msIntoSlot := in.ReceivedAt.Sub(slotStartTime).Milliseconds()
-	duration := time.Since(startTime)
+	payloadInfoChan := make(chan *common.VersionedPayloadInfo, 1)
 
-	go s.sendPayloadStats(in.Payload, log, true, payloadInfo, in.ReceivedAt, startTime, slotStartTime, msIntoSlot, id, in.ClientIP, in.ValidatorID, in.AccountID, latency, in.Cluster, in.UserAgent, in.SlotUID)
+	// validate and  fetch payload from cache
+	go func(ctx context.Context, l zerolog.Logger, parent trace.Span) {
+		ctx, childSpan := s.tracer.Start(ctx, "validateAndFetchPayload")
+		defer childSpan.End()
 
-	*log = log.With().
-		Dur("duration", duration).
-		Int64("slotStartTime", slotStartTime.UnixMilli()).
-		Int64("msIntoSlot", msIntoSlot).
-		Str("blockValue", payloadInfo.BlockValue).
-		Logger()
+		payloadInfo, err := s.validateAndFetchPayload(ctx, blindedBeaconBlock)
+		if payloadInfo != nil {
```

Comment:
we need a err check here, sometimes validateAndFetchPayload returns non-nil and err